sdk/servicebus/Azure.Messaging.ServiceBus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,309 @@
+
+For more information on Service Bus error scenarios and recommended ways to resolve them, please see the [Service Bus Messaging Exceptions][ServiceBusMessagingExceptions] guidance.
+
+
+### Find information about a ServiceBusException
+
+The exception includes some contextual information to assist in understanding the context of the error and its relative severity.  These are:
+
+- `EntityPath` : Identifies the Service Bus entity from which the exception occurred, if available.
+
+- `IsTransient` : Indicates whether or not the exception is considered recoverable. In the case where it was deemed transient, the appropriate retry policy has already been applied and all retries were unsuccessful.
+
+- `Message` : Provides a description of the error that occurred and relevant context.
+
+- `StackTrace` : Represents the immediate frames of the call stack, highlighting the location in the code where the error occurred.
+
+- `InnerException` : When an exception was the result of a service operation, this will often be a `Microsoft.Azure.Amqp.AmqpException` instance describing the error, following the [OASIS AMQP 1.0 spec][AmqpSpec].
+
+- `Reason` : Provides a set of well-known reasons for the failure that help to categorize and clarify the root cause. These are intended to allow for applying exception filtering and other logic where inspecting the text of an exception message wouldn't be ideal. Some key failure reasons are:
+
+  - **Service Timeout** : This indicates that the Service Bus service did not respond to an operation within the expected amount of time. This may have been caused by a transient network issue or service problem. The Service Bus service may or may not have successfully completed the request; the status is not known. In the case of accepting the next available session, this exception indicates that there were no unlocked sessions available in the entity.
+
+  - **Quota Exceeded** : This typically indicates that there are too many active receiver operations for a single entity. In order to avoid this error, reduce the number of potential concurrent receives. You can use batch receives to attempt to receive multiple messages per receive request. Please see [Service Bus quotas][ServiceBusQuotas] for more information.
+
+  - **Message Size Exceeded** : This indicates that the max message size has been exceeded. For standard tier namespaces, a single message may not exceed 256KB. For premium tier namespaces, the max message size is 100MB. This includes the body of the message, as well as any associated metadata and system overhead. The best approach for resolving this error is to reduce the number of messages being sent in a batch or the size of the body included in the message. Because size limits are subject to change, please refer to [Service Bus quotas][ServiceBusQuotas] for specifics.  
```

Comment:
I'm a bit averse to putting server-side limits in _our_ documentation if we can avoid it, mostly because if the doc is referred to by commit ID or copied then it can become misleadingly stale. 

Do you know if either of those will be the case here? Do we link to the troubleshooting guide in a way that it's always going to be the absolute latest in git, and not just some copy, or even a fixed commit ID?

Reply:
I share these concerns and don't believe that we can guarantee how the guide is linked/used.   I'm also uncomfortable with having to remember to tweak this for any service changes.